python/equistore-operations/equistore/operations/one_hot.py

```diff
@@ -60,15 +59,19 @@ def one_hot(labels: Labels, dimension: Labels) -> np.ndarray:
 
     name = dimension.names[0]
 
-    indices = np.zeros(len(labels), dtype=np.int64)
-    for i, entry in enumerate(labels[name]):
-        position = dimension.position([entry])
+    indices = _dispatch.zeros_like(dimension.values, [len(labels)])
+    labels_name = labels.column(name)
+    for i in range(labels_name.shape[0]):
+        entry = labels_name[None, i]
+        position = dimension.position(entry)
```

Comment:
instead of looping through everything in Python, we should also be able to rewrite this in term of `Labels.union`/`Labels.intersection`

Reply:
I tried, but the issue is that we would have to create a `Labels` by extracting a single dimension of `labels`, and that leads to repeated entries.

Reply:
I'll now try with array/tensor operations

Reply:
Update: I can't see an obvious way that doesn't require a lot of dispatch code and makes the error messages a lot more cryptic. I would stick with the current implementation until the operation becomes a bottleneck in our models